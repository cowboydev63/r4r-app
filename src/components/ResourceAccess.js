import React from 'react';
import PropTypes from 'prop-types';
import { Theme } from '../theme';
import SVG from './SVG';

// The API returns resources with one of three types of resourceAccess.type keys. This 
// let's us create the corresponding title client-side.
const resourceAccessTitles = {
    'open': 'This resource is free',
    'register': 'This resource requires registration',
    'cost': 'This resource requires payment'
}

const ResourceAccess = ({
    type,
    notes,
}) => (
    <article aria-label="Resource Access Information">
        <Theme element="div" className='resource__access'>
            <SVG iconType={ type }/>
            <h4>{ resourceAccessTitles[type] }</h4>
            {
                notes ? <p>{ notes }</p> : null
            }
        </Theme>
    </article>
);

ResourceAccess.propTypes = {
    type: PropTypes.string.isRequired,
    notes: PropTypes.string,
}

export default ResourceAccess;