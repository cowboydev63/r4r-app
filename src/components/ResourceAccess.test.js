import React from 'react';
import { shallow } from 'enzyme';
import ResourceAccess from './ResourceAccess';

describe('ResourceAccess Component', () => {
  it('renders without crashing', () => {
    shallow(<ResourceAccess type="open" />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<ResourceAccess type="open" />);
    expect(wrapper).toMatchSnapshot();
  })

})