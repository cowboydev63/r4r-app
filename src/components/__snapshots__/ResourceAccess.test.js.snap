// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceAccess Component renders correctly 1`] = `
<article
  aria-label="Resource Access Information"
>
  <Theme
    className="resource__access"
    element="div"
  >
    <SVG
      className="r4r-svg-container"
      iconType="open"
    />
    <h4>
      This resource is free
    </h4>
  </Theme>
</article>
`;
